src/rhub/api/policies.py

```diff
@@ -0,0 +1,161 @@
+            new[key] = value
+    return new
+
+
+def check_access(func):
+    """
+    Check whether user is in policy owners group
+    """
+    def inner(*args, **kwargs):
+        keycloak = get_keycloak()
+        if 'id' in kwargs:
+            current_user = kwargs['user']
+            group_name = 'policy-' + str(kwargs['id']) + '-owners'
+            group_list = keycloak.user_group_list(current_user)
+            groups = {group['name']: group for group in group_list}
+            if group_name in groups.keys():
+                # User has access to delete/edit policy
+                return func(*args, **kwargs)
+            else:
+                # User does not have access to delete/edit policy
+                return problem(403, 'Forbidden', 'You do not own this policy')
+        else:
+            return func(*args, **kwargs)
+    return inner
+
+
+def list_policies(user):
+    '''
+    API endpoint to provide the constraints from the policy table
+    '''
+    policies = model.Policy.query.all()
+    return [denormalize(row2dict(policy)) for policy in policies]
+
+
+def create_policy(user, body):
+    """
+    API endpoint to create a policy (JSON formatted)
+    """
+    body = normalize(body)
+    keycloak = get_keycloak()
+    groups = {group['name']: group for group in keycloak.group_list()}
+    current_user = get_current_user(user)[0]
+    policy = model.Policy(**body)
+    db.session.add(policy)
+    db.session.commit()
+    group_name = 'policy-' + str(row2dict(policy)['id']) + '-owners'
+    groups[group_name] = keycloak.group_create({'name': group_name})
+    keycloak.group_user_add(current_user['id'], groups[group_name])
+    keycloak.group_role_add('policy-owner', groups[group_name])
+    return denormalize(row2dict(policy))
+
+
+def search_policies(user, body):
+    """
+    API endpoint to list/search policies for attributes (UUID/Dept/Name*)
+    """
+    body = normalize(body)
+    sql_query = model.Policy.query
+    for key, value in body.items():
+        sql_query.filter(getattr(model.Policy, key).like(value))
+    policies = sql_query.all()
+    return [denormalize(row2dict(policy)) for policy in policies]
+
+
+def get_policy(user, id):
+    """
+    API endpoint to get policy by id
+    """
+    policies = model.Policy.query.filter(model.Policy.id == id)
+    return [denormalize(row2dict(policy)) for policy in policies]
+
+
+@check_access
+def update_policy(user, id, body):
+    """
+    API endpoint to update policy attributes
+    """
+    body = normalize(body)
+    try:
+        model.Policy.query.filter(model.Policy.id == id).update(body)
+        db.session.commit()
+        updated_record_row = model.Policy.query.filter(model.Policy.id == id).one()
+        updated_record_dict = row2dict(updated_record_row)
+    except exc.NoResultFound:
+        return problem(404, 'Not Found', 'Record Does Not Exist')
+    return denormalize(updated_record_dict)
+
+
+@check_access
+def delete_policy(user, id):
```

Comment:
`policy-...-owner` group in Keycloak can be also deleted. Group for a policy that doesn't exist will be useless I guess.